portality/models/bibjson.py

```diff
@@ -41,6 +41,11 @@ def title(self, val):
     ## work with the identifiers
 
     def add_identifier(self, idtype, value):
+        all = self.get_identifiers()
+        existing = [x for x in all if x["type"] == idtype]
+        if existing:
+            for x in existing:
+                all.remove(x)
```

Comment:
This is a good idea to check, and actually there's some underlying code which will do this for you.  Instead of doing this manually, just use

`self._delete_from_list("identifier", matchsub=idobj)`